src/gui/sdl_gui.cpp

```diff
@@ -1960,22 +1960,13 @@ static bool maybe_autoswitch_shader()
 		return false;
 	}
 
-	// The shaders need a canvas size as their target resolution
-	const auto canvas_size_px = sdl.renderer->GetCanvasSizeInPixels();
-
-	if (canvas_size_px.IsEmpty()) {
-		return false;
-	}
```

Comment:
Is this a case we need to handle? Your new code does not handle this.

Reply:
It's not strictly necessary (can't happen), but in the final state `maybe_autoswitch_shader()` will have morphed into `RENDER_NotifyVideoModeChanged()` which in turn will call the `NotifyVideoModeChanged()` method, and there's an existing assert for an empty canvas:

```cpp
void OpenGlRenderer::NotifyVideoModeChanged(const VideoMode& video_mode)
{
	const auto canvas_size_px = GetCanvasSizeInPixels();

	// We always expect a valid canvas and DOS video mode
	assert(!canvas_size_px.IsEmpty());
	assert(video_mode.width > 0 && video_mode.height > 0);

```